teps/0128-scheduled-runs.md

```diff
@@ -0,0 +1,279 @@
+---
+status: implementable
+title: Scheduled Runs
+creation-date: '2022-12-20'
+last-updated: '2022-12-20'
+authors:
+- '@vdemeester'
+- '@sm43'
+- '@lbernick'
+collaborators: []
+---
+
+# TEP-0128: Scheduled Runs
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+- [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes allowing Task/Pipeline users to schedule runs on a recurring basis.
+
+## Motivation
+
+Currently, to get a run to occur on a regular basis, a user must write a cron job that either creates the run itself
+(which duplicates much of triggers' functionality) or pings an EventListener.
+If the EventListener is exposed outside of the cluster, the user must also sign the cron job event body and verify it in a custom interceptor
+(although it's unlikely a user would choose to expose an ingress for an EventListener used only for cron jobs).
+
+### Goals
+
+- Can create new PipelineRuns, TaskRuns, or CustomRuns on a recurring basis without creating a webhook or writing a CronJob
+
+### Non-Goals
+
+- Polling a repository (or other external system) for changes; this is covered in [TEP-0083](./0083-polling-runs-in-tekton.md).
+
+### Use Cases
+
+- Release Pipeline: User would like to create releases of their project on a nightly basis.
+
+## Proposal
+
+Add a `schedule` field following Kubernetes CronJob syntax to the EventListener CRD, for example:
+
+```yaml
+kind: EventListener
+spec:
+  schedule: "0 0 * * *"
```

Comment:
What's the reasoning for adding schedule at the eventListener level vs at the Trigger level?  At the EL level, wouldn't this mean that each trigger in that EL would be fired at the same time?

Reply:
Moved this to alternative solutions, and noted that users may not want all the triggers to fire at the same time.